pennylane/ftqc/lattice.py

```diff
@@ -0,0 +1,189 @@
+# Copyright 2025 Xanadu Quantum Technologies Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# pylint: disable=protected-access, inconsistent-return-statements
+"""
+This file defines classes and functions for creating lattice objects that store topological
+connectivity information for measurement-based quantum computing (MBQC) and fault-tolerant
+quantum computing (FTQC).
+"""
+
+from typing import Dict, List, Union
+
+import networkx as nx
+
+
+class Lattice:
+    """Represents a qubit lattice for measurement-based quantum computing (MBQC) and fault-tolerant quantum computing (FTQC).
```

Comment:
```suggestion
    """Represents a given lattice structure, with a given connectivity and dimensionality.
```

Same as above, we don't need MBQC/FTQC awareness here. The lattice is used to define a structure we can use elsewhere.